products/cms-integration/debugging-field-guide.md

```diff
@@ -0,0 +1,144 @@
+and a message like the following will be logged to the console:
+
+    node.2bddb1a7-6fb1-4503-838d-9c2fcb51c46a (node-health_care_region_page) is invalid after transformation:
+    {
+      "keyword": "type",
+      "dataPath": ".entity.fieldLinkFacilityEmergList",
+      "schemaPath": "#/properties/entity/properties/fieldLinkFacilityEmergList/type",
+      "params": {
+        "type": "object,null"
+      },
+      "message": "should be object,null"
+    }
+    Data found at .entity.fieldLinkFacilityEmergList: []
+
+From this message we can see the following:
+
+-   The exact entity that failed validation
+    -   `node.2bddb1a7-6fb1-4503-838d-9c2fcb51c46a`
+-   Which content model it was
+    -   `node-health_care_region_page`
+-   Which schema failed
+    -   &ldquo;invalid after transformation&rdquo; means the schema in question can be found in
+        `schemas/transformed/node-health_care_region_page.js`
+    -   &ldquo;invalid before transformation&rdquo; would mean the schema could be found in
+        `schemas/raw/node-health_care_region_page.js`
+-   The field that failed validation
+    -   Look in the `dataPath`
+-   The schema rule that was violated
+    -   `schemapath` tells us what rule it was, `params` tells us what the valid
+        options were
+        -   Technically, it tells us what the parameters were for that particular rule
+    -   In this case, we can see it was the `fieldLinkFacilityEmergList`&rsquo;s `type`
+        which was supposed to be either an object or null
+-   The exact error message returned from [AJV](https://github.com/epoberezkin/ajv)
+    -   Look in `message`
+    -   This is the easy way to figure out what the error was, but it&rsquo;s sometimes
+        still mystifying
+-   What the actual data was that failed the validation
+    -   In this case, we can see `.entity.fieldLinkFacilityEmergList` was an empty
+        array
+    -   The schema expected it to be either an object or null, so this failure makes
+        sense
+
+
+<a id="orgf97a4c8"></a>
+
+### Solution
+
+Armed with this information, we can crack open the transformer and see what&rsquo;s
+going on.
+
+**Note:** If a validation error occurs while running the unit tests, you may have
+to scroll up quite a ways to see the actual validation error output. Just look
+for the yellow / gold writing.
+
+
+<a id="org830b381"></a>
+
+## Object doesn&rsquo;t deep equal object??
+
+Ideally, we&rsquo;d catch all the validation errors locally, but that won&rsquo;t always be
+the case. Unfortunately, the error output from Mocha is sometimes limitted in
+Jenkins. As such, when the unit test encounter an error, it logs the entire test
+file and the output of the transformation for manual comparison.
+
+
+<a id="orgb85397c"></a>
+
+### Solution
+
+Open the full logs in Jenkins to find the error output. Search for
+
+    Transformed entity in the test JSON file:
+
+
+<a id="org4eae7ce"></a>
+
+## TODO Transformer trouble
```

Comment:
What kinds of content do you think will go here?

Reply:
Right. No idea. I was hoping you could help. :sweat_smile: 

Otherwise, we can remove it entirely.